packages/cli/src/rpc/explain/explain.ts

```diff
@@ -208,15 +213,35 @@ export class Explain extends EventEmitter {
   }
 }
 
-const codeSelections = new LRUCache<string, string>({
+const codeSelections = new LRUCache<string, string | ContextV2.UserContextItem[]>({
   maxSize: 1024 ** 2 * 10 /* 10 MB */,
-  sizeCalculation: (value) => value.length,
+  sizeCalculation: (value) => {
+    if (typeof value === 'string') {
+      return value.length;
+    }
+    const size = value.reduce((a, v) => {
+      let ret = v.type.length;
+      switch (v.type) {
+        case ContextV2.ContextItemType.CodeSelection:
+          ret += v.location!.length + v.content!.length;
+          break;
+        case ContextV2.ContextItemType.CodeSnippet:
+          ret += v.content!.length;
+          break;
+        case ContextV2.ContextItemType.File:
+          ret += v.location!.length;
```

Comment:
We shouldn't need to use the non-null assertion operator, it generally indicates something is mis-typed.

Consider something like:
```ts
interface CodeSelectionContextItem {
    type: 'code-selection';
    location: string;
    content: string;
}

interface CodeSnippetContextItem {
    type: 'code-snippet';
    content: string;
    location?: never;
}

// This doesn't include the full list
type UserContextItem = CodeSelectionContextItem | CodeSnippetContextItem;
```